src/pages/VenuePage/VenuePage.tsx

```diff
@@ -115,92 +90,47 @@ const VenuePage: React.FC = () => {
     currentTimestamp >
       event.start_utc_seconds + event.duration_minutes * ONE_MINUTE_IN_SECONDS;
 
-  const isUserVenueOwner = user && venue?.owners?.includes(user.uid);
+  const isUserVenueOwner = userId && venue?.owners?.includes(userId);
   const isMember =
     user && venue && isUserAMember(user.email, venue.config?.memberEmails);
 
-  // Camp and PartyMap needs to be able to modify this
-  // Currently does not work with roome
-  const location = venueName;
-  useLocationUpdateEffect(user, venueName);
+  // NOTE: User location updates
 
-  const newLocation = { [location]: getCurrentTimeInMilliseconds() };
-  const isNewLocation = profile?.lastSeenIn
-    ? !profile?.lastSeenIn[location]
-    : false;
+  const updateUserLocationToCurrentVenue = useCallback(() => {
+    if (!userId || !venueName) return;
 
-  const newLocations = {
-    ...prevLocations,
-    ...newLocation,
-  };
+    updateLocationData(userId, {
+      [venueName]: getCurrentTimeInMilliseconds(),
+    });
+  }, [userId, venueName]);
+
+  useInterval(() => {
+    updateUserLocationToCurrentVenue();
+  }, LOC_UPDATE_FREQ_MS);
 
   useEffect(() => {
-    if (
-      user &&
-      location &&
-      isNewLocation &&
-      ((!unmounted && !retainAttendance) || retainAttendance) &&
-      (!profile?.lastSeenIn || !profile?.lastSeenIn[location])
-    ) {
-      updateLocationData(
-        user,
-        location ? newLocations : prevLocations,
-        profile?.lastSeenIn
-      );
-      setUnmounted(false);
-    }
-    if (
-      user &&
-      profile &&
-      (profile.lastSeenIn === null || profile?.lastSeenIn === undefined)
-    ) {
-      updateLocationData(
-        user,
-        location ? newLocations : prevLocations,
-        profile?.lastSeenIn
-      );
-    }
-  }, [
-    isNewLocation,
-    location,
-    newLocation,
-    newLocations,
-    prevLocations,
-    profile,
-    retainAttendance,
-    unmounted,
-    user,
-  ]);
+    updateUserLocationToCurrentVenue();
+  }, [updateUserLocationToCurrentVenue]);
 
   useEffect(() => {
-    const leaveRoomBeforeUnload = () => {
-      if (user && !retainAttendance) {
-        const locations = { ...prevLocations };
-        delete locations[venueName];
-        setUnmounted(true);
-        updateLocationData(user, locations, undefined);
-      }
-    };
-    window.addEventListener("beforeunload", leaveRoomBeforeUnload, false);
-    return () => {
-      window.removeEventListener("beforeunload", leaveRoomBeforeUnload, false);
-    };
-  }, [prevLocations, retainAttendance, user, venueName]);
+    if (!userId) return;
+
+    // NOTE: Clear user location on page close
+    window.addEventListener("beforeunload", () =>
+      updateLocationData(userId, {})
```

Comment:
If I have 2 pages open and close one of them, isn't this going to clear both locations?

Reply:
Actually looks like, you are keeping only 1 location open. Have you talked with Ed about this, because the very first implementation was like this and we had to change it to support multiple venues.

Reply:
Agreed. This is an edge case. But I think this implementation is better than the one we have now

Reply:
I asked @sparkleed about this and he said it is OK

Reply:
The 'implementation detail' of clearing this data by updating it with an empty object doesn't belong in individual components. There should be a function in the same file as `updateLocationData` called something like `clearLocationData` that handles this.